sympy/stats/crv.py

```diff
@@ -240,10 +240,56 @@ def characteristic_function(self, t, **kwargs):
                 return cf
         return self.compute_characteristic_function(**kwargs)(t)
 
+    @cacheit
+    def compute_moment_generating_function(self, **kwargs):
+        """ Compute the moment generating function from the PDF
+
+        Returns a Lambda
+        """
+        x, t = symbols('x, t', real=True, finite=True, cls=Dummy)
+        pdf = self.pdf(x)
+        mgf = integrate(exp(t * x) * pdf, (x, -oo, oo))
+        return Lambda(t, mgf)
+
+    def _moment_generating_function(self, t):
+        return None
+
+    def moment_generating_function(self, t, **kwargs):
+        """ Moment generating function """
+        if len(kwargs) == 0:
+
+            try:
+                mgf = self._moment_generating_function(t)
+                if mgf is not None:
+                    return mgf
+            except NotImplementedError:
+                return None
+        return self.compute_moment_generating_function(**kwargs)(t)
+
     def expectation(self, expr, var, evaluate=True, **kwargs):
         """ Expectation of expression over distribution """
-        integral = Integral(expr * self.pdf(var), (var, self.set))
-        return integral.doit(**kwargs) if evaluate else integral
+        if evaluate:
+            try:
+                p = poly(expr, var)
+
```

Comment:
The number of blank lines could be reduced.

(PEP 8: Use blank lines in functions, sparingly, to indicate logical sections.)